### PR TITLE
Added missing volume fields and fixed snapshot for performace type

### PR DESCRIPTION
**Pull request by pramodgavali**
1. Added missing fields username,BackendIPaddress,FileNetworkMountaddress
2. Created volume from snapshot of  performance volume type was not working - fixed

### CODE REVIEW

File: volume-providers/softlayer/block/block_volume_manager.go [hunk 1]
```diff
@@ -258,11 +258,11 @@ func (sls *SLBlockSession) VolumeCreateFromSnapshot(snapshot provider.Snapshot,
 	originalStorageType := *originalVolume.StorageType.KeyName
 	isPerformanceVolume := false
 	finalPrices := []datatypes.Product_Item_Price{}
+	volumeCategory := fmt.Sprintf(`storage_%s`, "block")
```
Comment (arahamad) [inline volume-providers/softlayer/block/block_volume_manager.go]: Can you please check `volumeCategory` for endurance and performance are different, Can you please verify for both endurance and performance storage

File: volume-providers/softlayer/file/file_volume_manager.go [hunk 2]
```diff
@@ -263,11 +263,11 @@ func (sls *SLFileSession) VolumeCreateFromSnapshot(snapshot provider.Snapshot, t
 	originalStorageType := *originalVolume.StorageType.KeyName
 	isPerformanceVolume := false
 	finalPrices := []datatypes.Product_Item_Price{}
+	volumeCategory := fmt.Sprintf(`storage_%s`, "file")
```
Comment (arahamad) [inline volume-providers/softlayer/file/file_volume_manager.go]: same here

File: lib/provider/data_types.go [hunk 3]
```diff
@@ -70,7 +79,7 @@ type Volume struct {
 	// LunID the lun of volume
 	LunID string `json:"lunId,omitempty"`
 
-	// TargetIPAddresses list of target IP addresses for iscsi
+	// TargetIPAddresses list of target IP addresses for iscsi. Applicable for block storage only
```
Comment (arahamad) [inline lib/provider/data_types.go]: For this variable can you add prefix as `Block` as this is only for block, similarly you used `File` for FileNetworkMountAddress.

File: volume-providers/softlayer/block/block_volume_manager.go (hunk 1 above)
Comment (pramodgavali) [reply volume-providers/softlayer/block/block_volume_manager.go]: I tested code for both performace and endurance

File: lib/provider/data_types.go (hunk 3 above)
Comment (arahamad) [reply lib/provider/data_types.go]: can you please address following 
1- I think `Block` will be more readable as compare to Iscsi
2- Please aline comments as per variable i.e line number 82

**Pull request merged**